src/Microsoft.AspNet.Mvc.Core/OptionDescriptors/OptionDescriptorExtensions.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.AspNet.Mvc.OptionDescriptors;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Extension methods for collections of option descriptors.
+    /// </summary>
+    public static class OptionDescriptorExtensions
+    {
+        /// <summary>
+        /// Returns the only instance of <typeparamref name="TInstance"/> from a sequence of option descriptors.
+        /// </summary>
+        /// <typeparam name="TInstance">The type of the instance to find.</typeparam>
+        /// <param name="descriptors">A sequence of <see cref="IOptionDescriptor{object}"/>.</param>
+        /// <returns>The only instance of <typeparamref name="TInstance"/> from a sequence of 
+        /// <see cref="IOptionDescriptor{object}"/>.</returns>
+        /// <exception cref="InvalidOperationException"> 
+        /// Thrown if there is not exactly one <typeparamref name="TInstance"/> in the sequence.</exception>
+        public static TInstance InstanceOf<TInstance>([NotNull] this IEnumerable<IOptionDescriptor<object>> descriptors)
+        {
+            var instance = descriptors.Single(d => d.OptionType == typeof(TInstance)
+                && d.Instance != null).Instance;
+            return (TInstance)instance;
+        }
+
+        /// <summary>
+        /// Returns the only instance of <typeparamref name="TInstance"/> from a sequence of option descriptors, 
+        /// or a default value if the sequence is empty.
```

Comment:
huh?  the relevant documentation is for `SingleOrDefault<TSource>(this IEnumerable<TSource>, Func<TSource, Boolean>)` (with my highlighting):
> Returns the only element of a sequence that satisfies a specified condition or a default value __if no such element exists__; this method throws an exception if more than one element satisfies the condition.